lua/neogit/buffers/common.lua

```diff
@@ -25,22 +27,45 @@ M.Diff = Component.new(function(diff)
   }, { foldable = true, folded = false, context = true })
 end)
 
--- Use vim iter api?
 M.DiffHunks = Component.new(function(diff)
   local hunk_props = vim
     .iter(diff.hunks)
     :map(function(hunk)
-      hunk.content = vim.iter(diff.lines):slice(hunk.diff_from + 1, hunk.diff_to):totable()
+      local header = diff.lines[hunk.diff_from]
+      local content = vim.list_slice(diff.lines, hunk.diff_from + 1, hunk.diff_to)
+      local job = nil
+      if config.values.log_pager ~= nil then
+        job = Job:new {
+          command = config.values.log_pager[1],
+          args = vim.list_slice(config.values.log_pager, 2),
+        }
+        job:start()
+        for _, part in ipairs { diff.header, { header }, content } do
+          for _, line in ipairs(part) do
+            job:send(line .. "\n")
+          end
+        end
+        job.stdin:close()
+      end
```

Comment:
Couple notes:
- This isn't really the right place for this - I would consider components part of the "view" layer, and as such really shouldn't be doing any IO. I know that's not strictly adhered to in the codebase, but I didn't come to this conclusion until about a year ago when I redid the status buffer. So I'd recommend putting this in `neogit.lib.git.diff`
- Unless you have a good reason not to use `vim.system()` here, prefer that over plenary.job. Fewer external deps is always better IMO.